src/Agent/NewRelic/Agent/Core/Config/BootstrapConfiguration.cs

```diff
@@ -56,5 +134,55 @@ private bool CheckServerlessModeEnabled(configuration localConfiguration)
             // fall back to config file
             return localConfiguration.serverlessModeEnabled;
         }
+
+        private void SetAgentEnabledValues()
+        {
+            _agentEnabledWithProvenance = TryGetAgentEnabledFromWebConfig();
+            if (_agentEnabledWithProvenance != null)
+            {
+                return;
+            }
+
+            _agentEnabledWithProvenance = TryGetAgentEnabledFromAppSettings();
+            if (_agentEnabledWithProvenance != null)
+            {
+                return;
+            }
+
+            _agentEnabledWithProvenance = new ValueWithProvenance<bool>(_agentEnabledValueFromLocalConfig, ConfigurationFileName);
+        }
+
+        private ValueWithProvenance<bool> TryGetAgentEnabledFromWebConfig()
+        {
+            return TryGetAgentEnabledSetting(_getWebConfigSettingWithProvenance);
+        }
+
+        private ValueWithProvenance<bool> TryGetAgentEnabledFromAppSettings()
+        {
+            return TryGetAgentEnabledSetting(getStringValueWithProvenance);
+
+            ValueWithProvenance<string> getStringValueWithProvenance(string key)
+            {
+                return new ValueWithProvenance<string>(_configurationManagerStatic.GetAppSetting(key), _configurationManagerStatic.AppSettingsFilePath);
+            }
+        }
+
+        private ValueWithProvenance<bool> TryGetAgentEnabledSetting(Func<string, ValueWithProvenance<string>> getStringValueWithProvenance)
+        {
+            try
+            {
+                var stringValueWithProvenance = getStringValueWithProvenance(Constants.AppSettingsAgentEnabled);
+                if (stringValueWithProvenance != null && stringValueWithProvenance.Value != null && bool.TryParse(stringValueWithProvenance.Value, out var value))
+                {
+                    return new ValueWithProvenance<bool>(value, stringValueWithProvenance.Provenance);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error($"Failed to read {Constants.AppSettingsAgentEnabled} from local config.", ex);
```

Comment:
I think you have this syntax backwards: the exception should be the first parameter.

Reply:
Good catch! I didn't look closely enough at the copilot output.